src/core/server/saved_objects/migrations/actions/check_for_unknown_docs.ts

```diff
@@ -40,43 +40,46 @@ export const checkForUnknownDocs =
   ({
     client,
     indexName,
-    unusedTypesQuery,
+    excludeOnUpgradeQuery,
     knownTypes,
   }: CheckForUnknownDocsParams): TaskEither.TaskEither<
-    RetryableEsClientError | UnknownDocsFound,
-    {}
+    RetryableEsClientError,
+    UnknownDocsFound | {}
   > =>
   () => {
-    const query = createUnknownDocQuery(unusedTypesQuery, knownTypes);
-
+    const query = createUnknownDocQuery(excludeOnUpgradeQuery, knownTypes);
     return client
       .search<SavedObjectsRawDocSource>({
         index: indexName,
         body: {
+          size: 1000,
           query,
```

Comment:
We would have to add the same aggregation you added in `src/core/server/saved_objects/deprecations/unknown_object_types.ts` to this action too.

The deprecations unknown documents query isn't 100% the same as the `checkForUnknownDocs` action because the deprecations query doesn't have the must_nots from the `REMOVED_TYPES` constant and `calculateExcludeFilters`.

The deprecations query only runs when upgrade assistant is used and for now it's only enabled in 7.17 (we hope to enable it for minors going forward). So its is actually dead code. I'll create an issue for us to follow-up so that we fix the deprecations query before we re-enable upgrade assistant in minors. For this PR I don't think it's worth fixing this (also it doesn't really impact users except that it gives a false deprecation that needs to be actioned).

Reply:
Sorry I didn't get your point! In my [latest commit](https://github.com/elastic/kibana/pull/132167/commits/a23db47b38536f93f596b40822b93eaafc99c808) I refactored the code a bit, so that both `check_for_unknown_docs.ts` and `unknown_object_types.ts` use the same logic to perform the aggregated search and to extract the results.